wolfcrypt/src/pkcs12.c

```diff
@@ -269,15 +280,39 @@ static int GetSafeContent(WC_PKCS12* pkcs12, const byte* input,
     XMEMCPY(safe->data, input + localIdx, size);
     *idx = localIdx;
 
+    localIdx = 0;
+    input = safe->data;
+    size = safe->dataSz;
+
+#ifdef ASN_BER_TO_DER
+     if (pkcs12->der) {
```

Comment:
Why are we checking the field 'der' for not NULL.
(Please explicitly check for not NULL.)

We should only be converting is the first sequence has indefinite length.
If one item is indefinite, then all constructed ASN.1 items around it will be indefinite.

If the data that is encrypted is indefinite length, then the surrounding ASN.1 items don't need to be.
I believe the actual encrypted data is indefinite length, in the case you are looking at, so the constructed items will be indefinite length.

Reply:
For definite length, the raw input var `data` is parsed while `pkcs12->der` is parsed for indefinite length. It is the BER to DER converted form of `data`. The `der` field would only be initialized if indefinite length is encountered (see `d2i_PKCS12()`). That is rather vague though so I've added an `indefinite` field.